timemachine/cpp/src/barostat.cu

```diff
@@ -82,7 +83,7 @@ MonteCarloBarostat<RealType>::MonteCarloBarostat(
     cudaSafeMalloc(&d_volume_scale_, 1 * sizeof(*d_volume_scale_));
     cudaSafeMalloc(&d_volume_delta_, 1 * sizeof(*d_volume_delta_));
 
-    gpuErrchk(cudaMemset(d_volume_scale_, 0, 1 * sizeof(*d_volume_scale_)));
+    gpuErrchk(cudaMemset(d_volume_scale_, initial_volume_scale_factor, 1 * sizeof(*d_volume_scale_)));
```

Comment:
This needs to be a cudaMemcpy, cudaMemset is on raw bytes so this can become some random value.

https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__MEMORY.html#group__CUDART__MEMORY_1gf7338650f7683c51ee26aadc6973c63a

Reply:
Might be good to add a test for this as well

Reply:
Good catch! Added a failing assertion in https://github.com/proteneer/timemachine/pull/1151/commits/5d161e4af77e6e6f3053af6abb43b9a25c75719a and fixed in https://github.com/proteneer/timemachine/pull/1151/commits/94225e266d3ef3fe1cca0f985431af4d2c7da1a0